### PR TITLE
Merge 2.9 into develop

### PR DESCRIPTION
**Pull request by manadart**
Merge from 2.9 to bring forward:
- #14007 from wallyworld/update-golang-deps
- #14013 from hmlanigan/show-controller-panic
- #14012 from hmlanigan/remove-unit-subordinate
- #13998 from turrisxyz/setup-permissions
- #14011 from ycliuhw/update-ck-overlay
- #14010 from tlm/juju-jenkins-tests
- #14004 from tlm/juju-jenkins-tests
- #14003 from manadart/2.9-close-unit-loggers
- #14001 from jack-w-shaw/JUJU-1050_show_unit_life
- #13890 from juju/dependabot/github_actions/actions/upload-artifact-3
- #14000 from wallyworld/use-go1.18
- #13999 from tlm/juju-jenkins-tests
- #13991 from ycliuhw/fix/lp1968643
- #13997 from tlm/cross-compile
- #13931 from arnodel/juju-885-remove-modelaccess-connection-iface-method
- #13992 from jack-w-shaw/JUJU-1034_reduce_over_verbose_logging_network
- #13993 from hmlanigan/fix-intermittent-fail-testenqueuedoperation
- #13990 from wallyworld/handle-azure-quota-errors
- #13989 from benhoyt/start-worker-if-dead
- #13987 from hpidcock/filter-test-packages

Conflicts:
- cmd/juju/controller/showcontroller.go
- go.mod
- go.sum
- provider/azure/environ_test.go

### CODE REVIEW

**Comment by hmlanigan** [comment]
@manadart I'm unable to change the target branch from 2.9 to develop for you.  Looking to review.

**Review by hmlanigan — APPROVED** [review]
I reviewed the showcontroller.go conflict, small edit. Ran go mod tidy. Hopefully the unit tests will show if there any issues with the azure test change noted as conflicts,

**Comment by hmlanigan** [comment]
$$merge$$

**Review by tlm — APPROVED** [review]
lgtm happy that my changes are represented correctly.

**Comment by barrettj12** [comment]
Feel free to ignore the Smoke, Snapcraft, Upgrade and macOS client tests - I'm [working on fixing these](https://github.com/juju/juju/pull/14006). The Ubuntu client tests and Jenkins failures are probably cause for concern though.

File: provider/azure/environ_test.go [hunk 1]
```diff
@@ -856,13 +869,69 @@ func (s *environSuite) TestStartInstanceCommonDeployment(c *gc.C) {
 			`common resource deployment status is "Failed"`)
 }
 
+func (s *environSuite) TestStartInstanceCommonDeploymentStorageAccount(c *gc.C) {
```
Comment (wallyworld) [inline provider/azure/environ_test.go:872]: delete me - this is gone in 3.0

File: provider/azure/environ_test.go [hunk 2]
```diff
@@ -856,13 +869,69 @@ func (s *environSuite) TestStartInstanceCommonDeployment(c *gc.C) {
...
+	s.assertStartInstanceRequests(c, s.requests[1:], assertStartInstanceRequestsParams{
+		imageReference: &xenialImageReference,
+		diskSizeGB:     32,
+		osProfile:      &s.linuxOsProfile,
+		instanceType:   "Standard_A1",
+		publicIP:       true,
+	})
+}
+
+func (s *environSuite) TestStartInstanceCommonDeploymentWithStorageAccountAndAvailabilitySetName(c *gc.C) {
```
Comment (wallyworld) [inline provider/azure/environ_test.go:898]: delete me - this is gone in 3.0

**Review by wallyworld — CHANGES_REQUESTED** [review]

**Comment by wallyworld** [comment]
Also, the `TestManifoldsDependenciesCAAS` error is because the `restore-watcher` worker is gone in 3.0 so somehow that has leaked back in with this PR

File: cmd/jujud/agent/machine/manifolds_test.go [hunk 3]
```diff
@@ -1093,6 +1103,513 @@ var expectedMachineManifoldsWithDependencies = map[string][]string{
...
+		"controller-port",
+		"http-server-args",
+		"is-controller-flag",
+		"lease-manager",
+		"model-cache",
+		"model-cache-initialized-flag",
+		"model-cache-initialized-gate",
+		"multiwatcher",
+		"raft-transport",
+		"restore-watcher",
```
Comment (wallyworld) [inline cmd/jujud/agent/machine/manifolds_test.go:1142]: delete me - gone in 3.0

File: cmd/jujud/agent/machine/manifolds_test.go [hunk 4]
```diff
@@ -1093,6 +1103,513 @@ var expectedMachineManifoldsWithDependencies = map[string][]string{
...
```
Comment (wallyworld) [inline cmd/jujud/agent/machine/manifolds_test.go:1206]: delete me - gone in 3.0

File: cmd/jujud/agent/machine/manifolds_test.go [hunk 5]
```diff
@@ -1093,6 +1103,513 @@ var expectedMachineManifoldsWithDependencies = map[string][]string{
...
+		"is-controller-flag",
+		"state",
+		"state-config-watcher",
+	},
+
+	"restore-watcher": {
+		"agent",
+		"state",
+		"state-config-watcher",
+	},
```
Comment (wallyworld) [inline cmd/jujud/agent/machine/manifolds_test.go:1506]: delete me - gone in 3.0

**Review by wallyworld — CHANGES_REQUESTED** [review]

**Review by wallyworld — APPROVED** [review]

**Review by ycliuhw — APPROVED** [review]
My changes are correct, thank you

**Comment by manadart** [comment]
$$merge$$

**Pull request merged**